### PR TITLE
Make get dependencies callback region aware

### PR DESCRIPTION
**Pull request by georgifarashev**
**Description**

Changes proposed in this pull request:
- Make get dependencies callback region aware

**Related issue(s)**

- n/a

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [x] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [x] Mocks are regenerated, using the automated script

### CODE REVIEW

**Review by DimitarPetrov — DISMISSED** [review]

**Review by nkanchev — APPROVED** [review]

**Review by DimitarPetrov — APPROVED** [review]

**Review by nkanchev — APPROVED** [review]

**Review by krasish — APPROVED** [review]

File: components/director/internal/domain/tenant/repository.go [hunk 1]
```diff
@@ -370,6 +356,30 @@ func (r *pgRepository) GetLowestOwnerForResource(ctx context.Context, resourceTy
 	return dest.TenantID, nil
 }
 
+func (r *pgRepository) GetBySubscribedRuntimes(ctx context.Context) ([]*model.BusinessTenantMapping, error) {
```
Comment (DimitarPetrov) [inline components/director/internal/domain/tenant/repository.go]: Why is this needed? It seems unused?

File: components/director/internal/domain/tenant/repository.go [hunk 2]
```diff
@@ -370,6 +356,30 @@ func (r *pgRepository) GetLowestOwnerForResource(ctx context.Context, resourceTy
...
+	persist, err := persistence.FromCtx(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var entityCollection tenant.EntityCollection
+
+	query := `
```
Comment (DimitarPetrov) [inline components/director/internal/domain/tenant/repository.go]: The query should use the lister with the respective criteria. 

Also, the names of the fields and the tables should be parametrized accordingly (with the names in the file above and the ones for tenant accesses tables)

See: https://github.com/kyma-incubator/compass/blob/3e8dee4ba319971e9694d22680115e81e7059f39/components/director/pkg/resource/resource.go#L118

 and 

https://github.com/kyma-incubator/compass/blob/5a8eaeb471b31f97fb2d0e79a3286b323d76c9da/components/director/internal/repo/tenant_access_m2m.go#L18

**Review by nkanchev — DISMISSED** [review]

**Review by desislavaa — APPROVED** [review]

**Review by desislavaa — DISMISSED** [review]

**Review by desislavaa — APPROVED** [review]

**Review by nkanchev — APPROVED** [review]
